flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/fanout/FanOutShardSubscriber.java

```diff
@@ -452,7 +477,7 @@ public void onComplete() {
         }
 
         private void cancelSubscription() {
-            if (cancelled) {
+            if (isCancelled()) {
```

Comment:
could we potentially leak some subscriptions here?

Reply:
_Should not_ happen, since the AWS SDK client `.close()` method is called during shutdown. I need to add this check here because we also saw `RejectedExecutionException` thrown from `subscription.cancel()` a few lines down